extensions/authentication-passport/README.md

```diff
@@ -181,46 +242,121 @@ class PassportBasicAuthProvider implements Provider<AuthenticationStrategy> {
   // You'd better define your strategy name as a constant, like
   // `const AUTH_STRATEGY_NAME = 'basic'`
   // You will need to decorate the APIs later with the same name
+  // Pass in the user profile factory
   convertToAuthStrategy(basic: BasicStrategy): AuthenticationStrategy {
-    return new StrategyAdapter(basic, AUTH_STRATEGY_NAME);
+    return new StrategyAdapter(
+      basic,
+      AUTH_STRATEGY_NAME,
+      this.myUserProfileFactory,
+    );
   }
 }
 ```
 
-2. Register the strategy provider
+2. Create a provider for the verify function.
 
-Register the strategy provider in your LoopBack application so that the
-authentication system can look for your strategy by name and invoke it:
+Here is an example provider named VerifyFunctionProvider which has a `value()`
+method that returns a function of type BasicVerifyFunction.
+
+```ts
+// In file 'providers/verifyfn.provider.ts'
+
+import {Provider} from '@loopback/core';
+import {repository} from '@loopback/repository';
+import {BasicVerifyFunction} from 'passport-http';
+import {INVALID_USER_CREDENTIALS_MESSAGE} from '../keys';
+
+export class VerifyFunctionProvider implements Provider<BasicVerifyFunction> {
+  constructor(@repository('users') private userRepo: MyUserRepository) {}
+
+  value(): BasicVerifyFunction {
+    const myThis = this;
+
+    return async function(username: string, password: string, cb: Function) {
+      let user: MyUser;
+
+      try {
+        //find user with specific username
+        const users: MyUser[] = await myThis.userRepo.find({
+          where: {username: username},
+        });
+
+        // if no user found with this username, throw an error.
+        if (users.length < 1) {
+          let error = new Error(INVALID_USER_CREDENTIALS_MESSAGE); //assign 401 in sequence
+          throw error;
+        }
+
+        //verify given password matches the user's password
+        user = users[0];
+        if (user.password !== password) {
+          let error = new Error(INVALID_USER_CREDENTIALS_MESSAGE); //assign 401 in sequence
+          throw error;
+        }
+
+        //return null for error, and the valid user
+        cb(null, user);
+      } catch (error) {
+        //return the error, and null for the user
+        cb(error, null);
+      }
+    };
+  }
+}
+```
+
+3. Register(bind) the providers
+
+Register the _verify function provider_ required by the strategy provider. Then
```

Comment:
I'm thinking whether we use: `_verify function provider_` or `VerifyFunctionProvider` (as shown in the code snippet below). Similarly for other instances.